dev/generate_headers.py

```
@@ -359,14 +360,19 @@ def needs_build(self):
                 return True, f"source file {source} is newer than destination"
         # Check source list if cachefile exists
         if self.cachefile.exists():
-            previous_sources = pickle.load(self.cachefile.open('rb'))['sorted_sources']
-            if sorted(self.sources) != previous_sources:
+            try:
+                previous_sources = pickle.load(self.cachefile.open('rb'))['sorted_sources']
```

Ruff (0.16.0)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)